PsfShimMonitor/MainWindow.xaml.cs

```diff
@@ -0,0 +1,720 @@
+                    string outputs = "";
+                    string caller = "";
+                    Int64 start = 0;
+                    Int64 end = 0;
+
+                    try
+                    {
+                        operation = (string)data.PayloadByName("Operation");
+                    }
+                    catch { }
+                    try
+                    {
+                        inputs = (string)data.PayloadByName("Inputs");
+                    }
+                    catch { }
+                    try
+                    {
+                        result = (string)data.PayloadByName("Result");
+                    }
+                    catch { }
+                    try
+                    {
+                        outputs = (string)data.PayloadByName("Outputs");
+                    }
+                    catch { }
+                    try
+                    {
+                        caller = (string)data.PayloadByName("Caller");
+                    }
+                    catch { }
+                    if (inputs == null & result == null && outputs == null)
+                    {
+                        try
+                        {
+                            outputs = (string)data.PayloadByName("Message");
+                        }
+                        catch { }
+                    }
+                    try
+                    {
+                        start = (Int64)data.PayloadByName("Start");
+                    }
+                    catch { }
+                    try
+                    {
+                        end = (Int64)data.PayloadByName("End");
+                    }
+                    catch { }
+                    EventItem ei = new EventItem((int)data.EventIndex,
+                                                    start,
+                                                    end,
+                                                    data.TimeStamp,
+                                                    data.ProcessName,  
+                                                    data.ProcessID,
+                                                    data.ThreadID,
+                                                    data.ProviderName,
+                                                    operation,
+                                                    inputs,
+                                                    result,
+                                                    outputs,
+                                                    caller
+                                                  );
+
+                    lock (_TEventListsLock)
+                    {
+                        _TEventListItems.Add(ei);
+                        AddToProcIDsList(data.ProcessID);
+                    }
+                    worker.ReportProgress((int)data.EventIndex);
+                };
+
+                res1 = myTraceEventSession.EnableProvider(etwp.guid);    
```

Comment:
What are res1 and res2 for?  Can you replace res1 and res2 with a name the describes what they are for?

Reply:
Event providers can be a little funky if app crashes occur.  These values are displayed in the status area of the window in the event of a problem.  Updated.